### DISCUSSION TITLE
IPF disagreement with TSL

### DISCUSSION THREAD

**Discussion by eugenianietov** [mtex-toolbox/mtex#1271] [Ask Anything]
Hi everyone,

I am trying to analyze a Bi (321 crystal symmetry) EBSD dataset acquired with a TSL-OIM system. I have a .ang file exported from TSL and loaded it into MTEX to do the analysis, but the results obtained with MTEX don't match with those obtained with TSL. Somehow the crystal directions seem to be pointing to another direction, so I guess it must be a disagreement between reference frames. I don't know if there is any fixed convention from TSL or something similar, it is the first time that I work with EBSD data acquired with a TSL system. You can see my mtex code below and the comparison to the TSL-OIM results.

```matlab
%% Import Script for EBSD Bi Data
%
% This script was automatically created by the import wizard. You should
% run the whole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.
run ~/Documents/mtex-5.7.0/startup_mtex.m

%% Specify File Names

% path to files
pname = '/Users/eugenianietov/Desktop/Samples_Purdue/Bi_Amey/2021_07_28_Sn-58Bi-Sample28 SEM';

% which files to be imported
fname = [pname '/Region 2-After Cleaning.ang'];

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('321', [4.5 4.5 12], 'X||a', 'Y||b*', 'Z||c*', 'mineral', 'Bismuth', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('422', [5.8 5.8 3.2], 'mineral', 'Tin', 'color', [0.56 0.74 0.56])};

setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Import the Data
% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame','setting 2');

% Set IPF ColorKey to TSL color key
ipfKey = ipfTSLKey(ebsd('Bi').CS);

% Plot scattered orientations in IPFZ
f1 = figure();
plot(ipfKey)
hold on
plotIPDF(ebsd('Bi').orientations,zvector,'markerSize',2,'points','all','marker','o','markeredgecolor','k')
hold off

% Color EBSD according to IPFZ legend
f2 = figure();
plot(ebsd('Bi'),ipfKey.orientation2color(ebsd('Bi').orientations)) 
legend('off')
```

The two obtained figures from mtex are the following:
![EBSD_Bi_IPFZ_mtex](https://user-images.githubusercontent.com/26228093/140331068-7a4b08d8-ee6a-4621-92f4-874a55fd11e1.png)
![IPFZ_Bi_mtex](https://user-images.githubusercontent.com/26228093/140331072-a09f54bb-22f6-4cdc-9353-f58793d08305.png)

The figures obtained with TSL are:
![EBSD_Bi_IPFZ_TSL](https://user-images.githubusercontent.com/26228093/140331924-0754e1e5-a608-4002-9414-352a8be289ad.png)
![IPFZ_Bi_TSL](https://user-images.githubusercontent.com/26228093/140331925-dd492fa2-8c38-4d3d-8f16-5034ad48fe37.png)

Any help or suggestion is welcome!! Thanks in advance :)

Eugenia

**Comment by kilir** [comment]
Hi,
yes, you should make sure you use the right reference frame.
Here's a good point to start:
https://mtex-toolbox.github.io/EBSDReferenceFrame.html
Hope that helps.
Cheers,
Rüdiger

**Comment by eugenianietov** [comment]
Hi,

Yes I have been trying to figure out which rotation I have to apply to the EBSD in order to get the TSL orientations, but for the moment nothing worked. In fact, the only trick that worked was to apply a reflection in the yvector to the ipf colorKey. This got the coloring exactly right, but obviously the orientations are still wrong. That's why I'm wondering if there is a fixed rotation due to the TSL convention or if this is not so straightforward... .

Cheers,
Eugenia

**Reply by kilir** [reply]
Mtex applies a rotation according to the "setting xy" you specify when importing your data. If you use "setting 2" on your OIM system, orientations should be imported properly. That would also imply that that x goes east and z into plane as you specified.
If you plot polefigures  (0001, 11-20 and 10-11) of your grain there, how do those compare with the ones you get from OIM (including A1 / A2 labels)?
Rüdiger

**Reply by eugenianietov** [reply]
Here you can see how the pole figures look without changing anything in my code:
Mtex:
![pf_mtex](https://user-images.githubusercontent.com/26228093/140413887-9b7f9e2e-0ca5-490d-8581-32ba3c94a1a8.png)

TSL:
![PF_TSL](https://user-images.githubusercontent.com/26228093/140414158-64a6117d-895b-4853-8192-a84be7636514.png)

I have already tried different rotations that make the pole figures match, but the IPF is not matching in any case (it changes but not that much...). Honestly, I am doing trial and error because I don't know if there's any proper way to do this.

Thanks for the help Rüdiger!
Eugenia

**Comment by lfgmorales** [comment]
Dear Eugenia,

If you have access to the SEM where you have done your measurements, check the reference frame for data acquisition (either in OIM DC, TEAM or APEX), although normally „setting 2“ is chosen (as written in your code), in your SEM you may have a different referwnce frame, and then if you choose the option „setting 2“, you will end up with different plots

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 4 Nov 2021, at 21:24, Eugenia Nieto Valeiras ***@***.***> wrote:

﻿

Here you can see how the pole figures look without changing anything in my code:
Mtex:
[pf_mtex]<https://user-images.githubusercontent.com/26228093/140413887-9b7f9e2e-0ca5-490d-8581-32ba3c94a1a8.png>

TSL:
[PF_TSL]<https://user-images.githubusercontent.com/26228093/140414158-64a6117d-895b-4853-8192-a84be7636514.png>

I have already tried different rotations that make the pole figures match, but the IPF is not matching in any case (it changes but not that much...). Honestly, I am doing trial and error because I don't know if there's any proper way to do this.

Thanks for the help Rüdiger!
Eugenia

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1271#discussioncomment-1591072>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4LOJZ7EWAQNY3DJS3UKL2WRANCNFSM5HLSPW6Q>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1271#discussioncomment-1591072", "url": "https://github.com/mtex-toolbox/mtex/discussions/1271#discussioncomment-1591072", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by eugenianietov** [reply]
Hi,

Since I have no access to the SEM reference frame yet, I played with the orientations to match the provided pole figures. My code is currently:
`run ~/Documents/mtex-5.7.0/startup_mtex.m

%% Specify File Names

% path to files
pname = '/Users/eugenianietov/Desktop/Samples_Purdue/Bi_Amey';

% which files to be imported
fname = [pname '/Region 2-After Cleaning.ang'];

%% Specify Crystal and Specimen Symmetries
% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('321', [4.5 4.5 12], 'X||a', 'Y||b*', 'Z||c*', 'mineral', 'Bismuth', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('422', [5.8 5.8 3.2], 'mineral', 'Tin', 'color', [0.56 0.74 0.56])};

% setMTEXpref('xAxisDirection','east');
% setMTEXpref('zAxisDirection','intoPlane');

%% Import the Data
% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame','setting 2');

ebsd = rotate(ebsd,rotation.byAxisAngle(xvector,180*degree));
ebsd = rotate(ebsd,rotation.byAxisAngle(zvector,-90*degree));

h = [Miller(0,0,0,1,ebsd('Bi').CS),Miller(1,0,-1,0,ebsd('Bi').CS),Miller(1,0,-1,1,ebsd('Bi').CS)];
f = figure();
plotPDF(ebsd('Bi').orientations,h,'points','all','MarkerSize',5,'MarkerColor','b','MarkerEdgeColor','k','grid')

%% Set IPF ColorKey
ipfKey = ipfTSLKey(ebsd('Bi').CS);

f1 = figure();
plot(ipfKey)
hold on
plotIPDF(ebsd('Bi').orientations,zvector,'markerSize',2,'points','all','marker','o','markeredgecolor','k')
hold off`
As you can see below, I managed to match the pole figures to the TSL ones, but the IPF is not matching anyway. Does that make sense? I don't know if this is a problem from the ipfTSLkey somehow or if the reference frame is still wrong.
![PF_rotated_mtex](https://user-images.githubusercontent.com/26228093/140535668-6276d1ef-838c-4288-9fd4-6b664e4ef5ec.png)
![IPFZ_rotated_mtex](https://user-images.githubusercontent.com/26228093/140535667-21f10df9-5d02-4065-b9f6-b81d713980c8.png)

The thing is that I want to make sure that the reference frame is fine so I can analyze the data safely. I am worried that this is not the case due to the IPF disagreement. If this is just a coloring issue, I can work with the proper group color key, but being sure that the crystal orientations are fine. I can provide my .ang if anyone wants to give it a try, feel free to ask.

Thanks a lot!!

Eugenia

**Reply by kilir** [reply]
Assuming that 1) you ticked the "match map setting button" in your OIM and 2) that you used "setting 2" in OIM, having ` setMTEXpref('xAxisDirection','east');` and `setMTEXpref('zAxisDirection','intoPlane');` should be fine to view both, maps and pfs in the same frame. However, in your mtex pf, you are looking at x to the north and z out of plane, in your import script that remains commented.

Otherwise, if you like, you can drop me a file.

Cheers,
Rüdiger

**Reply by eugenianietov** [reply]
Well I still have to check how the .ang file was exported and the SEM reference frame preferences... . You can check the .ang file in the following link:
[https://we.tl/t-HF4XLDhfFL](url)

Cheers,
Eugenia

**Comment by kilir** [comment]
I'd say `'setting 1'` produced the pf / ipf you had from your OIM. But apparently the colorkey used in the map you provided is different from the ipfTSLKey.

Hope that helps,
Rüdiger.

**Reply by eugenianietov** [reply]
The colorkey obtained from TSL is the following:
![Legend_TSL](https://user-images.githubusercontent.com/26228093/140551625-8c76af0d-4ef3-4a3e-98b9-fbea9853676c.png)

Eugenia
